Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+                                DWRITE_FONT_STYLE style,
+                                DWRITE_FONT_STRETCH stretch,
+                                float fontSize,
+                                IDWriteTextFormat** outTextFormat) {
+    // Get the direct write factory instance
+    ComPtr<IDWriteFactory> dwriteFactory;
+    HRESULT hr = _DWriteCreateFactoryInstance(&dwriteFactory);
+    if (FAILED(hr)) {
+        TraceError(TAG, L"Unable to create DWrite factory");
+        return hr;
+    }
+
+    return dwriteFactory
+        ->CreateTextFormat(fontFamilyName, NULL, weight, style, stretch, fontSize, __GetUserDefaultLocaleName().data(), outTextFormat);
+}
+
+/**
+ * Helper that creates a IDWriteFontFamily object for a given family name
+ */
+HRESULT _DWriteCreateFontFamilyWithName(CFStringRef familyName, IDWriteFontFamily** outFontFamily) {
+    ComPtr<IDWriteFactory> dwriteFactory;
+    THROW_IF_FAILED(_DWriteCreateFactoryInstance(&dwriteFactory));
+
+    ComPtr<IDWriteFontCollection> systemFontCollection;
+    THROW_IF_FAILED(dwriteFactory->GetSystemFontCollection(&systemFontCollection));
+
+    CFIndex familyNameLength = CFStringGetLength(familyName);
+    std::vector<UniChar> unicharFamilyName(familyNameLength + 1);
+    CFStringGetCharacters(familyName, CFRangeMake(0, familyNameLength), unicharFamilyName.data());
+
+    size_t fontFamilyIndex;
+    BOOL fontFamilyExists;
+
+    THROW_IF_FAILED(systemFontCollection->FindFamilyName(reinterpret_cast<const wchar_t*>(unicharFamilyName.data()),
+                                                         &fontFamilyIndex,
+                                                         &fontFamilyExists));
+
+    if (!fontFamilyExists) {
+        TraceError(TAG, L"Unable to find font family \"%ws\"", unicharFamilyName.data());
+        return E_INVALIDARG;
+    }
+
+    return systemFontCollection->GetFontFamily(fontFamilyIndex, outFontFamily);
+}
+
+/**
+ * Helper function that creates an IDWriteFontFace object for a given font name.
+ */
+HRESULT _DWriteCreateFontFaceWithName(CFStringRef name, IDWriteFontFace** outFontFace) {
+    // Parse the font name for font weight, stretch, and style
+    // Eg: Bold, Condensed, Light, Italic
+    DWRITE_FONT_WEIGHT weight = DWRITE_FONT_WEIGHT_NORMAL;
+    DWRITE_FONT_STRETCH stretch = DWRITE_FONT_STRETCH_NORMAL;
+    DWRITE_FONT_STYLE style = DWRITE_FONT_STYLE_NORMAL;
+
+    CFStringRef familyName;
+    _InitDWriteFontPropertiesFromName(name, &weight, &stretch, &style, &familyName);
+
+    if (!familyName) {
+        TraceError(TAG, L"Unable to find family for font name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return E_INVALIDARG;
+    }
+
+    ComPtr<IDWriteFontFamily> fontFamily;
+
+    if (FAILED(_DWriteCreateFontFamilyWithName(familyName, &fontFamily)) || !fontFamily) {
+        TraceError(TAG, L"Unable to find family for font name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return E_INVALIDARG;
+    }
```

Comment:
RETURN macro here?

in general you want to avoid blindly originating a new HR as that loses context when looking at logs. all of the RETURN_* macros trace and the bottom most one *should* be using some logging to originate the error. blind conversions etc make it confusing when looking at it later.